### PR TITLE
Fix Interop-Proxy Dependencies

### PR DESCRIPTION
**Pull request by senkevinli**

### CODE REVIEW

File: services/telemetry/src/util.js [hunk 1]
```diff
@@ -19,37 +19,37 @@ export function cleanupAfter(cleanup, limit, message) {
 
   let earlyCall = false;
   let earlyErr = null;
+  let cleanupObj = {};
```
Comment (oldmud0) [inline services/telemetry/src/util.js:22]: So in the end, why did this have to be done?

File: services/telemetry/src/util.js [hunk 2]
```diff
@@ -19,37 +19,37 @@ export function cleanupAfter(cleanup, limit, message) {
...
+        clearTimeout(waitTimeout);
+        cleanup();
+
+        if (err) reject(err);
+        else resolve(result);
+      };
+
+      if (earlyCall) cleanupObj.finish(earlyErr);
+    });
+    return await done;
```
Comment (oldmud0) [inline services/telemetry/src/util.js:51]: You should be able to do this:
```js
return await new Promise((resolve, reject) => { /* ... */ }
```

File: services/telemetry/src/util.js (hunk 1 above)
Comment (senkevinli) [reply services/telemetry/src/util.js:22]: In the case where, 
```
let wait = async () => await new Promise((resolve, reject) => {
    // Call the finish function if the timeout is reached.
    let waitTimeout = setTimeout(() => {
      finish(Error(timeoutMessage));
    }, limit);

    // Cancel the timeout, cleanup, and mark the operation as
    // complete.
    finish = (err, result) => {
      clearTimeout(waitTimeout);
      cleanup();

      if (err) reject(err);
      else resolve(result);
    };

    if (earlyCall) finish(earlyErr);
  });
return { finish, wait };
```
Overriding the "finish" function when "wait" is called doesn't actually update the function where "cleanupAfter" returned.  The variable is only updated in cleanupAfter's closure. I think that's why the tests were failing initially.

File: services/telemetry/src/util.js (hunk 2 above)
Comment (senkevinli) [reply services/telemetry/src/util.js:51]: I thought it was a bit easier to read factored out, but I can change it if you'd like.

File: services/telemetry/src/util.js (hunk 1 above)
Comment (oldmud0) [reply services/telemetry/src/util.js:22]: Oh, I see, because of pass-by-value.

**Pull request merged**